### PR TITLE
examples update: add env var validation, stricter tsconfig

### PR DESCRIPTION
**Pull request by KATT**
- Add `env.js`-file that uses zod to validate env vars in build + at runtime
- Validate `next.config.js`
- Add public runtime variables
- Add helper for isomorphic redirects

### CODE REVIEW

**Review by sachinraja — APPROVED** [review]
Looks good, it would be nice if `Redirect` and `publicRuntimeConfig` were used somewhere though

**Comment by KATT** [comment]
Does it make more sense now?

File: examples/next-prisma-starter/src/pages/_app.tsx [hunk 1]
```diff
@@ -81,16 +82,33 @@ export default withTRPC<AppRouter>({
...
+    if (ctx.redirectTo) {
+      // Do redirect if requested
       return {
-        status: clientErrors[0].data?.httpStatus ?? 500,
+        status: 303, // 303 See other
+        headers: {
+          location: ctx.redirectTo,
+        },
+      };
+    }
```
Comment (sachinraja) [inline examples/next-prisma-starter/src/pages/_app.tsx:96]: Looks better, but I'm still not sure why this is needed/useful

Comment (KATT) [reply examples/next-prisma-starter/src/pages/_app.tsx:96]: Removed for now.

**Review by sachinraja — APPROVED** [review]

**Pull request merged**